internal/httputil/httputil.go

```diff
@@ -5,30 +5,60 @@ import (
 	"crypto/x509"
 	"net/http"
 	"os"
+	"path/filepath"
+	"strings"
 	"time"
 )
 
-func BuildHTTPClient(caCert string) (*http.Client, error) {
-	httpClient := &http.Client{Timeout: 10 * time.Second}
-
-	if caCert != "" {
-		// tlsFileWatcher, err := certwatcher.New(caCert, "")
+func LoadCerts(caDir string) (string, error) {
+	if caDir == "" {
+		return "", nil
+	}
 
-		cert, err := os.ReadFile(caCert)
+	var certs []string
+	err := filepath.Walk(caDir, func(path string, info os.FileInfo, err error) error {
 		if err != nil {
-			return nil, err
+			return err
 		}
-		caCertPool := x509.NewCertPool()
-		caCertPool.AppendCertsFromPEM(cert)
-		tlsConfig := &tls.Config{
-			RootCAs:    caCertPool,
-			MinVersion: tls.VersionTLS12,
+		if info.IsDir() {
+			return nil
```

Comment:
Is the intention to recurse through all subdirectories? I'm tempted to suggest that we stick to finding files in a single named directory without recursing, but I'm curious if you have a specific reason.

To avoid recursing, we can return `filepath.SkipDir` here.
```suggestion
			return filepath.SkipDir
```

Reply:
No, it's intended to skip that directory. I didn't see this in any examples.

Reply:
This appears to skip `"."` as a directory, meaning, it skips the current directory... so the e2es fail.

Reply:
OK, found one example, and the example has a comparison with a directory name, so we'd need to do that to properly use `filepathSkipDir`; not sure it's worth it?

Reply:
Maybe switch over to `os.ReadDir()`, which makes the intent more clear anyways?

Reply:
will look